lib/profile/node.rb

```diff
@@ -17,7 +17,9 @@ def self.all
     end
 
     def self.find(hostname=nil)
-      all.find { |node| node.hostname == hostname }
+      search_node = all.find { |node| node.hostname == hostname }
+      raise "Node '#{hostname}' not found" unless search_node
```

Comment:
We rely on `find` returning `nil` if nothing is found elsewhere in Profile (see [here](https://github.com/openflighthpc/flight-profile/blob/6998d4da9efa2bdfed9008c5bdc42b0dcc74e4c9/lib/profile/commands/apply.rb#L24)). We want to return `nil` if nothing is found because that's the behaviour of `Enumerable#find` ([docs](https://apidock.com/ruby/Enumerable/find)).

Please could you replace this change with checks for a `falsey` return value? We'll likely only need a check in `commands/view` and `commands/clean`.

Reply:
I see, new version should still fix the issue while preserving the original functionality of `Node::find`